src/java.base/share/classes/com/sun/crypto/provider/RSACipher.java

```diff
@@ -272,14 +273,14 @@ private void init(int opmode, Key key, SecureRandom random,
         int n = RSACore.getByteLength(rsaKey.getModulus());
         outputSize = n;
         bufOfs = 0;
-        if (paddingType == PAD_NONE) {
+        if (Objects.equals(paddingType, PAD_NONE)) {
```

Comment:
Not sure if this is necessary, with the current impl, paddingType is always assigned with one of internal PAD_XXX constants. Did I miss something?

Reply:
You are correct. IJ ignored details of the implementation . `==` will work fine. I'll revert the change.